encoding/httpbinding/path_replace.go

```diff
@@ -22,33 +22,33 @@ func bufCap(b []byte, n int) []byte {
 // replacePathElement replaces a single element in the path []byte.
 // Escape is used to control whether the value will be escaped using Amazon path escape style.
 func replacePathElement(path, fieldBuf []byte, key, val string, escape bool) ([]byte, []byte, error) {
-	fieldBuf = bufCap(fieldBuf, len(key)+3) // { <key> [+] }
```

Comment:
I think some comments would make this clearer

// search for "{\<key\>}". If not found, search for the greedy version "{\<key\>+}". If none are found, return error

I'm also curious if this is a hot function or why we need to do this whole `bufCap` byte slice checking. This would be easier if we didn't do this weird thing

Reply:
I'm also confused why they repeatedly use the `bufCap` to do bytes modification, maybe just to save memory in some long path scenario

Reply:
just to expand, this could actually be hot function if you're calling it every time that you route a request that has an httpbinding. Unfortunately without some benchmark or profiling changing this may cause a regression, so I'm fine just leaving it as-is for now